src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiServiceImpl.scala

```diff
@@ -413,10 +415,24 @@ class EServiceApiServiceImpl(
       descriptor: CatalogDescriptor,
       updateEServiceDescriptorState: CatalogDescriptorState
     ): CatalogDescriptor = {
-      val activatedAt: Option[OffsetDateTime] =
-        if (descriptor.state == Draft && updateEServiceDescriptorState == Published) offsetDateTimeSupplier.get().some
-        else descriptor.activatedAt
-      descriptor.copy(state = updateEServiceDescriptorState, activatedAt = activatedAt)
+      (descriptor.state, updateEServiceDescriptorState) match {
+        case (Draft, Published)      =>
+          descriptor.copy(state = updateEServiceDescriptorState, publishedAt = offsetDateTimeSupplier.get().some)
+        case (Published, Suspended)  =>
+          descriptor.copy(state = updateEServiceDescriptorState, suspendedAt = offsetDateTimeSupplier.get().some)
+        case (Suspended, Published)  =>
+          descriptor.copy(state = updateEServiceDescriptorState, suspendedAt = None)
+        case (Suspended, Deprecated) =>
+          descriptor.copy(
+            state = updateEServiceDescriptorState,
+            suspendedAt = None,
+            deprecatedAt = offsetDateTimeSupplier.get().some
+          )
+        // case (Published, Archived) => Archived logic not yet implemented
```

Comment:
missing
- Published -> Archived
- Suspended -> Archived

Reply:
Archived state is actually a missing logic in frontend/backend. We can't deal with these cases right now, as far as I could understand.

Reply:
I see, but I suggest to implement the set of the timestamp anyway.
It is likely that when the logic will be implemented, we will forget this part and introduce a bug (it is easy to find where the Archived status is used, but it is far more difficult to find where it is NOT used).